rewrite-gradle/src/main/java/org/openrewrite/gradle/Assertions.java

```diff
@@ -17,27 +17,47 @@
 
 import org.intellij.lang.annotations.Language;
 import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
 import org.openrewrite.Parser;
 import org.openrewrite.SourceFile;
 import org.openrewrite.groovy.GroovyParser;
 import org.openrewrite.groovy.tree.G;
 import org.openrewrite.kotlin.KotlinParser;
 import org.openrewrite.kotlin.tree.K;
+import org.openrewrite.maven.MavenExecutionContextView;
+import org.openrewrite.maven.MavenParser;
+import org.openrewrite.maven.MavenSettings;
 import org.openrewrite.test.SourceSpec;
 import org.openrewrite.test.SourceSpecs;
 import org.openrewrite.test.UncheckedConsumer;
 import org.openrewrite.text.PlainText;
 import org.openrewrite.text.PlainTextParser;
+import org.openrewrite.xml.tree.Xml;
 
 import java.nio.file.Paths;
 import java.util.List;
 import java.util.function.Consumer;
 
+import static org.openrewrite.maven.tree.MavenRepository.MAVEN_LOCAL_DEFAULT;
+
 public class Assertions {
 
     private Assertions() {
     }
 
+    static void customizeExecutionContext(ExecutionContext ctx) {
+        MavenExecutionContextView mctx = MavenExecutionContextView.view(ctx);
+        boolean nothingConfigured = mctx.getSettings() == null &&
+                                    mctx.getLocalRepository().equals(MAVEN_LOCAL_DEFAULT) &&
+                                    mctx.getRepositories().isEmpty() &&
+                                    mctx.getActiveProfiles().isEmpty() &&
+                                    mctx.getMirrors().isEmpty();
+        if (nothingConfigured) {
+            // Load Maven settings to pick up any mirrors, proxies etc. that might be required in corporate environments
+            mctx.setMavenSettings(MavenSettings.readMavenSettingsFromDisk(mctx));
+        }
+    }
```

Comment:
Why this addition here? I'd aim to keep the changes small, and we've had no requests that I know of to load Maven settings for Gradle, and it seems odd.

Reply:
I'd revert this change to gradle/Assertions.java for now.

Reply:
> Why this addition here? I'd aim to keep the changes small, and we've had no requests that I know of to load Maven settings for Gradle, and it seems odd.

Because Gradle also uses the settings.xml file (and thus also uses the MavenPomDownloader). To be honest, I was kind of surprised this `customizeExecutionContext` only exists for Maven. This does not make any sense to me, as both Gradle and Maven uses the same artifactories etc.

Reply:
That may be, but since this was reported as an issue running tests specifically for Maven let's minimize the changes here. We do not expect our Gradle users to have a requirement of using particular corporate `~/.m2/settings.xml`, nor do we want to suggest they add one for now.